keras2onnx/parser.py

```diff
@@ -482,20 +482,28 @@ def _get_output_nodes(node_list, node):
     return [n_ for n_ in node_list if n_ not in nodes_has_children]  # need to keep the order.
 
 
+def _filter_out_input(node_name):
+    # tf.keras BN layer sometimes create a placeholder node 'scale' in tf2.x. 
+    # Given bn layer will be converted in a whole layer, it's fine to just filter this node out.
+    filter_out = re.match(r"batch_normalization_\d+\/scale$", node_name)
+    filter_out = filter_out or re.match(r"batch_normalization_\d+\/cond/input", node_name) # since tf 2.2
+    return filter_out
+
+
 def _advance_by_input(cur_node, layer_nodes, subgraph, inputs, graph_inputs, q_overall):
     for input_ in cur_node.inputs:
         predecessor = input_.op
         if is_placeholder_node(predecessor):
-            # tf.keras BN layer sometimes create a placeholder node 'scale' in tf2.x.
-            # Given bn layer will be converted in a whole layer, it's fine to just filter this node out.
-            if not re.match(r"batch_normalization_\d+\/scale$", predecessor.name):
+            if not _filter_out_input(predecessor.name):
                 inputs.add(predecessor)
                 graph_inputs.add(predecessor)
+                continue
         if predecessor in layer_nodes or len(layer_nodes) == 0:
             subgraph.append(predecessor)
         else:
-            inputs.add(predecessor)
-            q_overall.put_nowait(predecessor)
+            if not _filter_out_input(predecessor.name):
```

Comment:
Can you confirm why some many filters needed?

Reply:
This seems redundant, removed.